datafusion/expr-common/src/signature.rs

```diff
@@ -125,6 +125,11 @@ pub enum TypeSignature {
     /// Fixed number of arguments of numeric types.
     /// See <https://docs.rs/arrow/latest/arrow/datatypes/enum.DataType.html#method.is_numeric> to know which type is considered numeric
     Numeric(usize),
+    /// Fixed number of arguments of string types.
+    /// The precedence of type from high to low is Utf8View, LargeUtf8 and Utf8.
+    /// Null is considerd as Utf8 by default
```

Comment:
Null is not Utf8, but may be coercible to that.
In fact, Null type should be coercible to any other type, but function signatue doesn't need to define that. The function signature needs to define what types the function _accepts_, and function resolution / expression analyzer should insert coercions as necessary.

Reply:
Yes, I agree `Null` should be coercable to any other type (and is generally). 

I agree the function signature should define what the function accepts

Reply:
If we found a function that reject null to string, we could define it with `user_define` or add another Signature::StringWithoutNull signature if there are more than one.